src/admin/__tests__/fetchTopicOffsets.spec.js

```diff
@@ -38,21 +39,35 @@ describe('Admin', () => {
       )
     })
 
-    test('returns the current topic offset', async () => {
+    const sendMessages = async n => {
       await admin.connect()
       await producer.connect()
 
-      const messages = Array(100)
+      const messages = Array(n)
         .fill()
         .map(() => {
           const value = secureRandom()
           return { key: `key-${value}`, value: `value-${value}` }
         })
 
       await producer.send({ acks: 1, topic: topicName, messages })
-      const offsets = await admin.fetchTopicOffsets(topicName)
+    }
 
+    test('returns the current topic offset', async () => {
+      await sendMessages(100)
+      const offsets = await admin.fetchTopicOffsets(topicName)
       expect(offsets).toEqual([{ partition: 0, offset: '100', low: '0', high: '100' }])
     })
+
+    test('returns the offsets from timestamp', async () => {
+      await sendMessages(10)
+      await sleep(10)
```

Comment:
What's the point of the sleep here? Whenever there's a sleep in a test, I get wary that there might be some flakiness introduced.

After the `send` is done, we should be able to use that timestamp, no?

Reply:
Yes. I'll remove the wait.